cookiejar.go

```diff
@@ -0,0 +1,221 @@
+package fasthttp
+
+import (
+	"bytes"
+	"net"
+	"sync"
+	"time"
+)
+
+// CookieJar manages cookie storage
+type CookieJar struct {
+	m           sync.Mutex
+	hostCookies map[string][]*Cookie
+}
+
+func (cj *CookieJar) init() {
+	cj.m.Lock()
+	{
+		if cj.hostCookies == nil {
+			cj.hostCookies = make(map[string][]*Cookie)
+		}
+	}
+	cj.m.Unlock()
+}
+
+func copyCookies(cookies []*Cookie) (cs []*Cookie) {
+	// TODO: Try to delete the allocations
+	cs = make([]*Cookie, 0, len(cookies))
+	for _, cookie := range cookies {
+		c := AcquireCookie()
+		c.CopyTo(cookie)
+		cs = append(cs, c)
+	}
+	return
+}
+
+// Get returns the cookies stored from a specific domain.
+//
+// If there were no cookies related with host returned slice will be nil.
+//
+// returned cookies can be released safely.
+func (cj *CookieJar) Get(uri *URI) (cookies []*Cookie) {
+	if uri != nil {
+		cookies = cj.get(uri.Host(), uri.Path())
+	}
+	return
+}
+
+func (cj *CookieJar) get(host, path []byte) (rcs []*Cookie) {
+	cj.init()
```

Comment:
I still don't like the fact that `cj.hostCookies` gets set here while it's a get method. I would prefer it would just return `nil` if `cj.hostCookies` is `nil`.